src/lib/Persistor.js

```diff
@@ -2,35 +2,50 @@
 
 var _ = require('underscore'),
     Q = require('q'),
-    Class = require('class.extend');
+    Class = require('class.extend'),
+    low = require('lowdb'),
+    FileSync = require('lowdb/adapters/FileSync'),
+    adapter, db;
 
 module.exports = Class.extend({
 
    init: function() {
-      // TODO: actually read/write to a file
-      this.inMemoryStorage = {};
+      adapter = new FileSync('./store/db.json');
+      db = low(adapter);
+      db.defaults({})
+         .write();
    },
 
    put: function(hash, range, value) {
-      if (!_.isObject(this.inMemoryStorage[hash])) {
-         this.inMemoryStorage[hash] = {};
+      var currentHash = db.get(hash).value(),
+          newObj = {};
+
+      if (!_.isObject(currentHash)) {
+         newObj = db.set(hash, {})
```

Comment:
I haven't looked into it much, but if `db.json` does not exist, the following error is thrown. It might be due to this code, not sure though.

```
./leaderboard-server/node_modules/lowdb/adapters/_stringify.js:5
  return JSON.stringify(obj, null, 2);
              ^

TypeError: Converting circular structure to JSON
    at Object.stringify (native)
    at FileSync.stringify (./leaderboard-server/node_modules/lowdb/adapters/_stringify.js:5:15)
    at FileSync.write (./leaderboard-server/node_modules/lowdb/adapters/FileSync.js:53:42)
    at LodashWrapper.db.write (./leaderboard-server/node_modules/lowdb/lib/main.js:37:21)
    at LodashWrapper.<anonymous> (./leaderboard-server/node_modules/lowdb/lib/main.js:19:15)
    at apply (./leaderboard-server/node_modules/lodash/lodash.js:474:27)
    at LodashWrapper.wrapper [as write] (./leaderboard-server/node_modules/lodash/lodash.js:5337:16)
    at Class.put (./leaderboard-server/src/lib/Persistor.js:32:11)
    at Object.<anonymous> (./leaderboard-server/src/test.js:4:11)
    at Module._compile (module.js:570:32)
```

Reply:
You were right. The app should now created db.json on start if the file doesnt exist.